airbyte-integrations/connectors/source-facebook-marketing/unit_tests/test_base_insight_streams.py

```diff
@@ -599,6 +599,74 @@ def test_start_date_with_lookback_window(
         stream.state = state
         assert stream._get_start_date().get("unknown_account").to_date_string() == expected_start_date
 
+    def test_start_date_when_monthly_time_increment(self, api, some_config):
+        """Should force start_date to be first day of the month when
+        time_increment is monthly"""
+
+        start_date = pendulum.datetime(2024, 1, 15)
+        end_date = pendulum.datetime(2024, 5, 25)
+        stream = AdsInsights(
+            api=api,
+            account_ids=some_config["account_ids"],
+            start_date=start_date,
+            end_date=end_date,
+            insights_lookback_window=28,
+            time_increment="monthly",
+        )
+        assert stream._get_start_date()["unknown_account"].to_date_string() == "2024-01-01"
```

Comment:
You have covered a lot a very useful behaviors in those test, thanks a lot! 

Can we update those tests to rely on `stream_slices`? Using the private method `_get_start_date`, I fear that:
* I might not be used where we think it is used and therefore doesn't define the behavior we expect
* When we will refactor without changing the behavior, those tests might also need to be updated while usually if the behavior doesn't change, the tests should stay as they are

I would also assume that the `test_generate_async_jobs` tests will not be needed anymore (or vice versa) as they seem to test the same thing, right?